File: Java/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.*;
class Solution {
    public int solution(int number, int limit, int power) {
        int[] divisor = getDivisor(number);
        for (int i = 0; i < divisor.length; i++) {
            if (divisor[i] > limit) divisor[i] = power;
        }
        return Arrays.stream(divisor).sum();
    }

    private int[] getDivisor(int number) {
        int[] divisor = new int[number];
        for (int i = 1; i <= number; i++) {
            int div = 0;
            for (int j = 1; j * j <= i; j++) {
                if (j * j == i) div++;
                else if (i % j == 0) div += 2;
            }
            divisor[i - 1] = div;
        }
        return divisor;
    }
}

File: Kotlin/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.kt
import kotlin.math.sqrt
class Solution {
    fun solution(number: Int, limit: Int, power: Int): Int {
        var answer: Int = 0

        for (i in 1..number) {
            var count = 0
            for (j in 1..sqrt(i.toDouble()).toInt()) {
                if (i % j == 0) {
                    count += 2
                }
                if (j * j == i) {
                    count -= 1
                }
            }
            answer += if (count > limit) power else count
        }

        return answer
    }
}